### PR TITLE
Use secure versions of memset and strncpy in libtrie and libcpuid

### LINKED ISSUE #9280: Third-party security reports vulnerabilities in libtrie and libcpuid
## Enhancement

1. Use of `memset` in https://github.com/pingcap/tiflash/blob/45eb5079cd6ff7e6fee91413c5c8abeaf57754a0/contrib/libbtrie/src/btrie.c#L49
2. Use of `strncpy` in https://github.com/pingcap/tiflash/blob/45eb5079cd6ff7e6fee91413c5c8abeaf57754a0/contrib/libcpuid/include/cpuid/cpuid_main.c#L288

### PR DESCRIPTION
**Pull request by zanmato1984**
### What problem does this PR solve?

Issue Number: close #9280

Problem Summary: see #9280.

### What is changed and how it works?

```commit-message

Replace the usages of `memset` and `strncpy` with their secure versions.

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Replace the usages of `memset` and `strncpy` with their secure versions.
```

### CODE REVIEW

**Review by JaySon-Huang — APPROVED** [review]

File: contrib/libbtrie/src/btrie.c [hunk 1]
```diff
@@ -46,7 +46,7 @@ btrie_create()
     tree->free  = NULL;
     tree->start = NULL;
     tree->size  = 0;
-    memset(tree->pools, 0, sizeof(btrie_t *) * BTRIE_MAX_PAGES);
+    memset_s(tree->pools, sizeof(char *) * BTRIE_MAX_PAGES, 0, sizeof(btrie_t *) * BTRIE_MAX_PAGES);
```
Comment (JaySon-Huang) [inline contrib/libbtrie/src/btrie.c:49]: The whole set of 'safe' functions with the _s suffixes is poorly supported. Clang/GCC does not support `memset_s`...
https://stackoverflow.com/a/50921096

Comment (JaySon-Huang) [reply contrib/libbtrie/src/btrie.c:49]: What about this?
```suggestion
#ifdef __STDC_LIB_EXT1__
    memset_s(tree->pools, sizeof(char *) * BTRIE_MAX_PAGES, 0, sizeof(btrie_t *) * BTRIE_MAX_PAGES);
#else
    memset(tree->pools, 0, sizeof(btrie_t *) * BTRIE_MAX_PAGES);
#endif
```

File: contrib/libcpuid/include/cpuid/cpuid_main.c [hunk 2]
```diff
@@ -285,7 +285,7 @@ static int cpuid_basic_identify(struct cpu_raw_data_t* raw, struct cpu_id_t* dat
 		brandstr[48] = 0;
 		i = 0;
 		while (brandstr[i] == ' ') i++;
-		strncpy(data->brand_str, brandstr + i, sizeof(data->brand_str));
+		strncpy_s(data->brand_str, BRAND_STR_MAX, brandstr + i, sizeof(data->brand_str));
```
Comment (JaySon-Huang) [inline contrib/libcpuid/include/cpuid/cpuid_main.c:288]:
```suggestion
#ifdef __STDC_LIB_EXT1__
		strncpy_s(data->brand_str, BRAND_STR_MAX, brandstr + i, sizeof(data->brand_str));
#else
		strncpy(data->brand_str, brandstr + i, sizeof(data->brand_str));
#endif
```

**Review by JaySon-Huang — CHANGES_REQUESTED** [review]
Failed to compile

File: contrib/libbtrie/src/btrie.c (hunk 1 above)
Comment (zanmato1984) [reply contrib/libbtrie/src/btrie.c:49]: The `_s` versions are basically Microsoft-only. Plus they are not as safe as they are designed to be. So I tend to not use them.

I'll close this PR.

**Comment by zanmato1984** [comment]
Closing as these are false-positives and the fix don't work for non-Windows platform at all.

**Pull request closed without merging**